pkg/controller/statusmanager/pod_status.go

```diff
@@ -135,9 +145,70 @@ func (status *StatusManager) SetFromPods() {
 		}
 	}
 
+	for _, ssName := range status.statefulSets {
+		ss := &appsv1.StatefulSet{}
+		if err := status.client.ClientFor(cnoclient.ManagementClusterName).CRClient().Get(context.TODO(), ssName, ss); err != nil {
```

Comment:
why not use `status.client.ClientFor(serviceItemFromRange.GetClusterName())`

Reply:
Also we might need to check if the result of `ClientFor` is not nil, just to be safe

Reply:
@kyrtapz I just realized this `status.statefulSets` is `types.NamespacedName` which doesn't have GetClusterName() interface.

Reply:
With the change as is we make an implicit assumption that statefulsets can only be deployed on the management cluster.
I am not sure if we should do that, since we are in control of `statefulSets` and the rest for watching objects maybe we could extend them to support multiple clusters? This could be another commit

Reply:
@kyrtapz I agree that `serviceItemFromRange.GetClusterName()` is better than making assumption on resource location. I will make another PR to fix it.

Reply:
Done